api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java

```diff
@@ -31,7 +48,107 @@
+        executeDataSet(XML_DATASET_PATH + XML_DATASET);
+
+        // initialize the service
+        providerManagementService = Context.getService(ProviderManagementService.class);
+    }
+    
+    @Test
+    public void testClearTimeComponent(){
+    	Date date = new Date(2018, 11, 15);
+    	date.setHours(10);
+    	date.setMinutes(32);
+    	date.setSeconds(46);
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(new Date(2018, 11, 15), result);
+    }
+    
+    @Test
+    public void testClearTimeComponentofNegativeDate(){
+    	Date date = new Date(1950, -1, 2);
+    	date.setHours(15);
+    	date.setMinutes(20);
+    	date.setSeconds(12);
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(new Date(1950, -1, 2), result);
+    }
+    
+    @Test
+    public void testClearTimeComponentofNewDay(){
+    	Date date = new Date(2018, 7, 20);
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(date, result);
+    }
 
+	@Test
+   	public void shouldSetupContext() {
+   		assertNotNull(Context.getService(ProviderManagementService.class));
+   	}
+    
+    @Test
+    public void checkWhenSupervisorListIsNotEmpty() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = Context.getPersonService().getPerson(6);
+    	Person supervisor = Context.getPersonService().getPerson(2);
+    	provider.setPerson(person);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	Assert.assertFalse(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    }
+    
+    @Test
+    public void checkWhenSupervisorListIsEmpty() throws PersonIsNotProviderException {
+    	Provider provider = new Provider();
+    	Person person = Context.getPersonService().getPerson(6);
+    	provider.setPerson(person);
+    	Assert.assertTrue(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    }
+    
+    @Test
+    public void checkForTheAccurateNumberOfSupervisorsAdded() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException, DateCannotBeInFutureException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = Context.getPersonService().getPerson(6);
+    	Person supervisor = Context.getPersonService().getPerson(2);
+    	provider.setPerson(person);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	Assert.assertEquals(1, ProviderManagementUtils.getSupervisors(provider).size());
+    }
+    
+    @Test
+    public void checkWhenSuperviseeListIsNotEmpty() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
```

Comment:
I'd make the same changes to the "supervisee list" methods that I suggest above for the "supervisor" list.